### PR TITLE
[FLINK-31598][table] Cleanup usage of deprecated TableEnvironment#registerTable

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

Replace the usage of deprecated `TableEnvironment#registerTable()` with `TableEnvironment#createTemporaryView()`

## Brief change log

- Replace the usage of deprecated `registerTable()` with `createTemporaryView()`
- Quotes the registered UUID table name in `TableTestBase` to fix the failed tests

## Verifying this change
This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by chucheng92 — COMMENTED** [review]
it's a good improvement. i checked and found 2 more places. 

SemiAntiJoinStreamITCase#testExistsWithUncorrelated_ComplexCondition
StreamTableEnvironmentImpl

**Comment by chucheng92** [comment]
@wuchong  hi, jark. Another question, do we have a timeline (e.g. 1.18) to remove these deprecate methods, otherwise it is still possible for implementors to call these methods. (In fact, currently there are too many deprecated interfaces or methods in flink, we can take a step at the table layer)

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/utils/TableTestBase.scala [hunk 1]
```diff
@@ -1185,7 +1185,7 @@ abstract class TableTestUtil(
       name)
     val operation = new RichTableSourceQueryOperation(identifier, tableSource, statistic)
     val table = testingTableEnv.createTable(operation)
-    testingTableEnv.registerTable(name, table)
+    testingTableEnv.createTemporaryView(name, table)
```
Comment (snuyanzin) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/utils/TableTestBase.scala:1188]: There is a TODO comment above mentioning `registerTable`. Shouldn't it be updated as well?

**Comment by snuyanzin** [comment]
+1 for this cleanup in general.
Also wonder about `SemiAntiJoinStreamITCase` and `StreamTableEnvironmentImpl` mentioned by @chucheng92

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/utils/TableTestBase.scala (hunk 1 above)
Comment (wuchong) [reply flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/utils/TableTestBase.scala:1188]: Good catch. I will update the comment.

**Review by chucheng92 — APPROVED** [review]
LGTM +1

**Review by snuyanzin — APPROVED** [review]

**Comment by wuchong** [comment]
@chucheng92 , yes, as I mentioned in the JIRA issue, this effort is to make it easier for us to remove the deprecated interfaces.

**Review by chucheng92 — APPROVED** [review]

**Pull request merged**